.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -7,3 +7,15 @@ resolves #1235
 ### This pullrequest changes
 
 <!-- Please describe what your pullrequest is changing -->
+
+<cut/>
+
+### Pull Request Readiness Checklist
+
+- [ ] I agree to contribute to the project under OpenCV ([BSD](https://github.com/opencv/opencv/blob/master/LICENSE)) license
+- [ ] The PR is proposed to proper branch ([details](https://github.com/opencv/opencv/wiki/How_to_contribute#before-you-start-contributing-you-should))
+- [ ] There is reference to original bug report and related work
+- [ ] There is test, if code has been changed
+- [ ] There is PR with opencv_extra changes, if any
+- [ ] There is performance test, if the change affects performance
+- [ ] The feature is well documented
```

Comment:
well-documented is too strong requirement, and quite subjective one. Probably, should be changed to

"If a new API is added, it should be documented using Doxygen-friendly comments"